src/QMCWaveFunctions/Fermion/DiracDeterminant.cpp

```diff
@@ -44,7 +44,11 @@ void DiracDeterminant<DU_TYPE>::set(int first, int nel, int delay)
 {
   FirstIndex = first;
   ndelay     = delay;
-  resize(nel, nel);
+
+  resize(nel, Phi->OrbitalSetSize);
+
+  if(Optimizable)
+    Phi->buildOptVariables(nel);
```

Comment:
I'm not sure if this is the best place to call this. Leave this message now so I don't forget.